metadata-ingestion/src/datahub_provider/_lineage_core.py

```diff
@@ -48,264 +49,57 @@ def send_lineage_to_datahub(
-                ),
-                *ownership_aspect,
-                *tags_aspect,
-            ],
-        )
-    )
+    hook = config.make_emitter_hook()
+    emitter = hook.make_emitter()
 
-    # exclude subdag operator tasks since these are not emitted, resulting in empty metadata
-    upstream_tasks = (
-        [
-            builder.make_data_job_urn_with_flow(flow_urn, task_id)
-            for task_id in task.upstream_task_ids
-            if dag.task_dict[task_id].subdag is None
-        ]
-        + upstream_subdag_task_urns
-        + upstream_subdag_triggers
+    dataflow = AirflowGenerator.generate_dataflow(
+        cluster=config.cluster,
+        dag=dag,
+        capture_tags=config.capture_tags_info,
+        capture_owner=config.capture_ownership_info,
     )
-
-    job_doc = (
-        (
-            operator.doc
-            or operator.doc_md
-            or operator.doc_json
-            or operator.doc_yaml
-            or operator.doc_rst
-        )
-        if not AIRFLOW_1
-        else None
+    dataflow.emit(emitter)
+    print(f"Emitted from Lineage: {dataflow}")
+
+    datajob = AirflowGenerator.generate_datajob(
+        cluster=config.cluster,
+        task=task,
+        dag=dag,
+        capture_tags=config.capture_tags_info,
+        capture_owner=config.capture_ownership_info,
     )
-
-    job_mce = models.MetadataChangeEventClass(
-        proposedSnapshot=models.DataJobSnapshotClass(
-            urn=job_urn,
-            aspects=[
-                models.DataJobInfoClass(
-                    name=task.task_id,
-                    type=models.AzkabanJobTypeClass.COMMAND,
-                    description=job_doc,
-                    customProperties=job_property_bag,
-                    externalUrl=job_url,
-                ),
-                models.DataJobInputOutputClass(
-                    inputDatasets=_entities_to_urn_list(inlets or []),
-                    outputDatasets=_entities_to_urn_list(outlets or []),
-                    inputDatajobs=upstream_tasks,
-                ),
-                *ownership_aspect,
-                *tags_aspect,
-            ],
+    datajob.inlets.extend(_entities_to_urn_list(inlets))
+    datajob.outlets.extend(_entities_to_urn_list(outlets))
+
+    datajob.emit(emitter)
+
+    if config.capture_executions:
+        dag_run: "DagRun" = context["dag_run"]
+
+        dpi = AirflowGenerator.run_datajob(
+            emitter=emitter,
+            cluster=config.cluster,
+            ti=ti,
+            dag=dag,
+            dag_run=dag_run,
+            datajob=datajob,
+            emit_templates=False,
         )
-    )
 
-    force_entity_materialization = [
-        models.MetadataChangeEventClass(
-            proposedSnapshot=models.DatasetSnapshotClass(
-                urn=iolet,
-                aspects=[
-                    models.StatusClass(removed=False),
-                ],
-            )
+        print(f"Emitted from Lineage: {dpi}")
```

Comment:
what is the recommended way to log statements in airflow?

Reply:
looks like `operator.log.info(..)` ?

Reply:
as far as I know, printing is a (the?) way to log